### ISSUE TITLE
cached route geometry is missing paths

### ISSUE THREAD

**Issue by markongithub** [code4jc/buswatcher#17]
It looks to me like the getRoutePoints.jsp endpoint only returns paths for trips that are currently in motion. (I've confirmed this behavior by requesting it for NJT route 13 at different times, hours apart.) So if we save the XML output and then later on encounter a bus that's on a different path ID, [this block](https://github.com/code4jc/buswatcher/blob/4044104e0433c9b66751dc2e7262e4c0aba1f907/buswatcher/lib/DataBases.py#L67) will not add any ScheduledStop instances for that trip. For routes with a lot of different branches, this can result in a large amount of lost data.

I can think of two possible solutions...

1. Add path_id as a parameter to [get_single_route_paths_and_coordinatebundle](https://github.com/code4jc/buswatcher/blob/4044104e0433c9b66751dc2e7262e4c0aba1f907/buswatcher/lib/TransitSystem.py#L94) and if we don't have that path ID in our geometry, try calling getRoutePoints.jsp again to see if it's available now. But it's not great because this cache failure will consistently happen several times each day.

2. Cache each path in its own file, and download getPatternPoints.jsp whenever we find one we haven't cached.

3. Try to load this data from the agency's GTFS bundle instead. It's less elegant because the ID numbers don't match anything returned from Clever Devices. But you get much greater consistency within the GTFS bundle.

Neither of my ideas is great. Have you encountered this problem, or have you been mainly looking at routes with more consistent paths?

Thanks!

**Comment by anthonymobile** [comment]
yes, the way i have modeled/parsing this is very hacky. #2 was how i planned to address this but never got to. #3 i agree is the ideal but im not sure how/if its possible to biuld a lookup table to convert clever devices trip ids to GTFS.
